client.go

```diff
@@ -966,15 +966,44 @@ func (f *File) readChunkAt(ch chan result, b []byte, off int64) (n int, err erro
 	return
 }
 
+func (f *File) readAtSequential(b []byte, off int64) (int, error) {
+	readed := 0
+
+	for {
+		endRead := int64(math.Min(float64(readed+f.c.maxPacket), float64(len(b))))
+		n, err := f.readChunkAt(nil, b[readed:endRead], off+int64(readed))
+		if n < 0 {
+			panic("sftp.File: returned negative count from readChunkAt")
+		}
+		if n > 0 {
+			readed += n
+		}
+		if err != nil {
+			if errors.Is(err, io.EOF) {
+				return readed, nil // return nil explicitly.
+			}
+			return readed, err
+		}
+		if readed == len(b) {
```

Comment:
We could easily make this part of the loop condition, rather than a subordinate `if` clause:
```
for read < len(b) {
  …
}

return read, nil
```